### PR TITLE
Make checkout js code easier to understand

### LINKED ISSUE #15074: Improve code readability for Javascript checkout behavior
This feedback should be handled for 1.7.7: https://github.com/PrestaShop/PrestaShop/pull/15042/files/2974dd7d78a422c10023120e5ed5e5482e9d74de

```
Consider extracting functions and objects, this code is very complicated to understand.

Picture this:

if (!clickedStep.isUnreachable()) {
	clickedStep.makeCurrent();
	if (clickedStep.hasContinueButton()) {
		steps.disableAllAfter(clickedStep)
	} else {
        steps.enableAllBefore(clickedStep)
    }
}

Looks better isn't it?
```

### PR DESCRIPTION
**Pull request by atomiix**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Refactoring checkout steps
| Type?         | refacto
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #15074
| How to test?  | See #15042. This PR modifies the code but the behavior should be the same as described in #15042 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16099)

### CODE REVIEW

File: themes/_core/js/checkout-steps.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+$.fn.isUnreachable = function () {
+  return this.hasClass('-unreachable');
+};
+
+$.fn.makeCurrent = function (step) {
+  this.removeClass('-current');
+  this.removeClass(currentStepClass);
+  step.toggleClass('-current');
```
Comment (PierreRambaud) [inline themes/_core/js/checkout-steps.js]:
```suggestion
  step.addClass('-current');
```

File: themes/_core/js/checkout-steps.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  step.toggleClass(currentStepClass);
```
Comment (PierreRambaud) [inline themes/_core/js/checkout-steps.js]:
```suggestion
  step.addClass(currentStepClass);
```

File: themes/_core/js/checkout-steps.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+const currentStepClass = 'js-current-step';
+const currentStepSelector = `.${currentStepClass}`;
+
+$.fn.step = function () {
+  this.off('click');
+  return this;
+};
+
+$.fn.getActiveSteps = function () {
```
Comment (PierreRambaud) [inline themes/_core/js/checkout-steps.js]: We couldn't have more than one active step :thinking:

Comment (atomiix) [reply themes/_core/js/checkout-steps.js]: I should rename this function, it's not "active", it's all the steps we already pass, including the current one.

File: themes/_core/js/checkout-steps.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+import $ from 'jquery';
+
+const currentStepClass = 'js-current-step';
+const currentStepSelector = `.${currentStepClass}`;
+
+$.fn.step = function () {
```
Comment (PierreRambaud) [inline themes/_core/js/checkout-steps.js]: Not sure adding all these functions to `$.fn` is a good idea. It should be in separated class or function like
`$.fn.checkout.step`  wdyt?

File: themes/_core/js/checkout-steps.js (hunk 3 above)
Comment (PierreRambaud) [reply themes/_core/js/checkout-steps.js]: Oh, so yes a renaming is needed ^^

File: themes/_core/js/checkout-steps.js (hunk 4 above)
Comment (Progi1984) [reply themes/_core/js/checkout-steps.js]: :+1: Because we can't see which steps these are : checkout ? registration ?...

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @atomiix

**Pull request merged**